source/postprocess/visualization/maximum_horizontal_compressive_stress.cc

```diff
@@ -47,134 +47,161 @@ namespace aspect
+                                                           (avg_compressive_stress *
                                                             orthogonal_directions[0]));
                   break;
-                }
 
-                // in 3d, use the formulas discussed in the
-                // documentation of the plugin below
                 case 3:
                 {
                   const double a = orthogonal_directions[0] *
-                                   (compressive_stress *
+                                   (avg_compressive_stress *
                                     orthogonal_directions[0]);
                   const double b = orthogonal_directions[1] *
-                                   (compressive_stress *
+                                   (avg_compressive_stress *
                                     orthogonal_directions[1]);
-                  const double c = 2.0*orthogonal_directions[0] *
-                                   (compressive_stress *
+                  const double c = 2.0 * orthogonal_directions[0] *
+                                   (avg_compressive_stress *
                                     orthogonal_directions[1]);
 
-                  // compute the two stationary points of f(alpha)
-                  const double alpha_1 = 1./2 * std::atan2 (c, a-b);
+                  const double alpha_1 = 0.5 * std::atan2(c, a - b);
                   const double alpha_2 = alpha_1 + numbers::PI/2;
 
-                  // then check the sign of f''(alpha) to determine
-                  // which of the two stationary points is the maximum
                   const double f_double_prime_1 = 2*(b-a)*std::cos(2*alpha_1)
                                                   - 2*c*std::sin(2*alpha_1);
                   double alpha;
                   if (f_double_prime_1 < 0)
                     alpha = alpha_1;
                   else
                     {
-                      Assert (/* f_double_prime_2 = */
-                        2*(b-a)*std::cos(2*alpha_2) - 2*c*std::sin(2*alpha_2) <= 0,
-                        ExcInternalError());
+                      Assert (2*(b-a)*std::cos(2*alpha_2) - 2*c*std::sin(2*alpha_2) <= 0,
+                              ExcInternalError());
                       alpha = alpha_2;
                     }
 
-                  // then re-assemble the maximum horizontal compressive_stress
-                  // direction from alpha and the two horizontal
-                  // directions
                   const Tensor<1,dim> n = std::cos(alpha) * orthogonal_directions[0] +
                                           std::sin(alpha) * orthogonal_directions[1];
 
-                  // finally compute the actual direction * magnitude,
-                  // now taking into account the pressure (with the
-                  // correct sign in front of the pressure for the
-                  // *compressive* stress)
```

Comment:
this part can of course be removed since you now include pressure from the onset